File: visa-dev-samplecode-master/vdp-ruby/test/cybersource/cybersource_payments_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class CybersourcePaymentsTest < Test::Unit::TestCase
  
  def setup
    @config = YAML.load_file('configuration.yml')
    @visa_api_client = VisaAPIClient.new
    @paymentAuthorizationRequest='''{
      "amount": "0",
      "currency": "USD",
      "payment": {
        "cardNumber": "4111111111111111",
        "cardExpirationMonth": "10",
        "cardExpirationYear": "2016"
       }
    }'''
  end
  
  def test_authorize_payments
    base_uri = 'cybersource/'
    resource_path = 'payments/v1/authorizations'
    api_key = @config['apiKey']
    response_code = @visa_api_client.doXPayTokenRequest(base_uri, resource_path, "apikey=#{api_key}", "Payment Authorization Request test", "post", @paymentAuthorizationRequest)
    assert_equal("201", response_code, "Payment Authorization test failed")
  end
end

File: visa-dev-samplecode-master/vdp-node/test/cybersource/cybersourcepaymentstest.js
var request = require('request');
var config = require('../../config/configuration.json');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var assert = require('chai').assert;
var req = request.defaults();
var randomstring = require('randomstring');

describe('Cybersource Payments Test', function() {
	var visaAPIClient = new VisaAPIClient();
	var paymentAuthorizationRequest = JSON.stringify({
		"amount": "0",
		"currency": "USD",
		"payment": {
			"cardNumber": "4111111111111111",
			"cardExpirationMonth": "10",
			"cardExpirationYear": "2016"
		}
	});

	it('Cybersource Payments Authorization Test',function(done){
		this.timeout(10000);
		var apiKey = config.apiKey;
		var baseUri = 'cybersource/';
		var resourcePath = 'payments/v1/authorizations';
		var queryParams = 'apikey=' + apiKey;
		visaAPIClient.doXPayRequest(baseUri, resourcePath, queryParams, paymentAuthorizationRequest, 'POST', {}, 
		function(err, responseCode) {
			if(!err) {
				assert.equal(responseCode, 201);
			} else {
				assert(false);
			}
			done();
		});
	});
});
